packages/components/src/DataList/components/DataListHeaderTile/DataListHeaderTile.tsx

```diff
@@ -40,6 +38,10 @@ export function DataListHeaderTile<T extends DataListObject>({
 
   const Tag = isSortable ? "button" : "div";
 
+  const selectedOption = sorting?.state
+    ? ({ ...sorting?.state } as SortableOptions)
+    : null;
```

Comment:
don't need to spread it and can actually take advantage of the `||` operator here. 

```suggestion
  const selectedOption: SortableOptions | null = sorting?.state || null;
```

adding the type instead of casting it is also safer

Reply:
Fair - just pushed this change